src/go/intelligence/utils/utils.go

```diff
@@ -16,10 +21,33 @@ import (
 //ConfigJSON variable
 var ConfigJSON models.Config
 
-//IfSilencedMap for storing ongoing silences
+//IfSilencedMap - variable for storing ongoing silences
 var IfSilencedMap map[string]int
 
-//ValidateURL function for constructing and validating AM URL
+//FirstRunSinceRestart - boolean variable for storing the information if the it's the first start of the service after restart or not
+var FirstRunSinceRestart bool
+
+//DashboardsCache - a cache for storing dashboards and Expiration time for updating the cache
+type DashboardsCache struct {
+	Dashboards models.AllDashboardsFetched
+	Expiration time.Time
+}
+
+//UpdateDashboardCache - function for updating the dashboards cache on expiration
+func (dCache *DashboardsCache) UpdateDashboardCache() {
+
+	if !FirstRunSinceRestart && !(dCache.Expiration.Equal(time.Now())) {
+		return
+	}
+
+	dCache.Dashboards = findDashboards()
+	dCache.Expiration = time.Now().Add(24 * ConfigJSON.AnnotationDashboard.DashboardsCacheExpiration * time.Hour)
```

Comment:
I think you should remove 24 factor here.

Reply:
Removed 24 factor. I was thinking we will have Expiration in terms of days, so 24.